### PR TITLE
Allowing comments in redirect input

### PR DESCRIPTION
**Pull request by jeff-allen-mongo**
Script will now ignore all lines in input beginning with '#'

### CODE REVIEW

File: mut/redirects/redirect_main.py [hunk 1]
```diff
@@ -89,219 +89,201 @@ def write_to_file(rules: List[RuleDefinition], output_path: str) -> None:
             f.write('\n')
 
 
+def parse_line(line, rc, line_num, version_regex, url_regex, dict_regex):
+    # strip \n from line
+    line = line.strip()
+
+    # allow for comments:
+    if (len(line) > 0 and line[0] != '#'):
```
Comment (i80and) [inline mut/redirects/redirect_main.py]: I'd probably restructure this condition:

```
if not line or line.startswith('#'):
    continue
...
```

File: mut/redirects/redirect_main.py [hunk 2]
```diff
@@ -89,219 +89,201 @@ def write_to_file(rules: List[RuleDefinition], output_path: str) -> None:
             f.write('\n')
 
 
+def parse_line(line, rc, line_num, version_regex, url_regex, dict_regex):
```
Comment (i80and) [inline mut/redirects/redirect_main.py]: Type annotations

```
from typing import Pattern
...
def parse_line(line: str, rc: RedirectContext, line_num: int,
               version_regex: Pattern, url_regex: Pattern, dict_regex: Pattern) -> None:
```

File: mut/redirects/redirect_main.py [hunk 3]
```diff
@@ -89,219 +89,201 @@ def write_to_file(rules: List[RuleDefinition], output_path: str) -> None:
...
+
+                value = value.strip()
+                value = rc.rule_substitute(value)
+                rc.add_definition(key, value)
+
+            # grab symlinks:
+            if keyword_split[0] == 'symlink':
+                type_split = line.split(':', 1)
+                sym_split = [sym.strip() for sym in type_split[1].split('->')]
+                alias = sym_split[0]  # did not exist previously
```
Comment (i80and) [inline mut/redirects/redirect_main.py]: alias, origin = sym_split

File: mut/redirects/redirect_main.py [hunk 4]
```diff
@@ -89,219 +89,201 @@ def write_to_file(rules: List[RuleDefinition], output_path: str) -> None:
...
+                # Process accordingly based on brackets in groups 1 and 4
+                if (match.group(2) not in rc.versions and match.group(2) != '*'):
+                    raise ValueError('ERROR in line {}: Version {} not present in version list'
+                                     .format(line_num, match.group(2)))
+                elif match.group(3):
+                    if (match.group(3) not in rc.versions and match.group(3) != '*'):
+                        raise ValueError('ERROR in line {}: Version {} not present in version list'
+                                         .format(line_num, match.group(3)))
+
+                    if (match.group(3) != '*'):
```
Comment (i80and) [inline mut/redirects/redirect_main.py]: Unnecessary parenthesis

File: mut/redirects/redirect_main.py [hunk 5]
```diff
@@ -89,219 +89,201 @@ def write_to_file(rules: List[RuleDefinition], output_path: str) -> None:
...
+                    version = match.group(2)
+                    if version == '*':
+                        for ver in rc.versions:
+                            rc.generate_rule(is_temp, ver, old_url, new_url)
+                    else:
+                        rc.generate_rule(is_temp, version, old_url, new_url)
+
+
 def parse_source_file(source_path: str, output: str) -> None:
+
```
Comment (i80and) [inline mut/redirects/redirect_main.py]: Nix this empty line

File: mut/redirects/redirect_main.py (hunk 1 above)
Comment (jeff-allen-mongo) [reply mut/redirects/redirect_main.py]: Fixed. Also moved this check to the parse_source_file() function to shave one more indentation.

File: mut/redirects/redirect_main.py (hunk 2 above)
Comment (jeff-allen-mongo) [reply mut/redirects/redirect_main.py]: Done. Also removing the dict_regex value / param because it doesn't actually seem to be used anywhere.

**Pull request merged**